fastify.js

```diff
@@ -585,6 +585,17 @@ function build (options) {
   function addHook (name, fn) {
     throwIfAlreadyStarted('Cannot call "addHook" when fastify instance is already started!')
 
+    // TODO: v3 instead of log a warning, throw an error
+    if (name === 'onSend' || name === 'preSerialization') {
+      if (fn.constructor.name === 'AsyncFunction' && fn.length === 4) {
```

Comment:
Can you point me to the place in the spec that says this will always be true when a function is created with the `async` keyword?

Reply:
This seems to be it: https://www.ecma-international.org/ecma-262/9.0/index.html#sec-async-function-constructor-properties

> - has a **name** property whose value is **"AsyncFunction"**.

Reply:
I need to learn more about how functions are “constructed”, but that seems definitive.